fluent-syntax/src/parser/ftlstream.rs

```diff
@@ -1,394 +1,211 @@
+use super::Result;
+use std::str;
+
+pub struct ParserStream<'p> {
+    pub source: &'p [u8],
+    pub length: usize,
+    pub ptr: usize,
 }
 
-static INLINE_WS: [char; 2] = [' ', '\t'];
-static SPECIAL_LINE_START_CHARS: [char; 4] = ['}', '.', '[', '*'];
-
-impl<I> FTLParserStream<I> for ParserStream<I>
-where
-    I: Iterator<Item = char>,
-{
-    fn skip_inline_ws(&mut self) {
-        while let Some(ch) = self.ch {
-            if !INLINE_WS.contains(&ch) {
-                break;
-            }
-            self.next();
-        }
-    }
-
-    fn peek_inline_ws(&mut self) {
-        while let Some(ch) = self.current_peek() {
-            if !INLINE_WS.contains(&ch) {
-                break;
-            }
-            self.peek();
-        }
-    }
-
-    fn skip_blank_lines(&mut self) {
-        loop {
-            self.peek_inline_ws();
-
-            if self.current_peek() == Some('\n') {
-                self.skip_to_peek();
-                self.next();
-            } else {
-                self.reset_peek(None);
-                break;
-            }
+impl<'p> ParserStream<'p> {
+    pub fn new(stream: &'p str) -> Self {
+        ParserStream {
+            source: stream.as_bytes(),
+            length: stream.len(),
+            ptr: 0,
         }
     }
 
-    fn peek_blank_lines(&mut self) {
-        loop {
-            let line_start = self.get_peek_index();
-
-            self.peek_inline_ws();
-
-            if self.current_peek_is('\n') {
-                self.peek();
-            } else {
-                self.reset_peek(Some(line_start));
-                break;
-            }
-        }
+    pub fn is_current_byte(&self, b: u8) -> bool {
+        self.source.get(self.ptr) == Some(&b)
     }
 
-    fn skip_indent(&mut self) {
-        self.skip_blank_lines();
-        self.skip_inline_ws();
+    pub fn is_byte_at(&self, b: u8, pos: usize) -> bool {
+        self.source.get(pos) == Some(&b)
     }
 
-    fn expect_char(&mut self, ch: char) -> Result<()> {
-        if self.ch == Some(ch) {
-            self.next();
-            return Ok(());
+    pub fn expect_byte(&mut self, b: u8) -> Result<()> {
+        if !self.is_current_byte(b) {
+            return error!(ErrorKind::ExpectedToken(b as char), self.ptr);
         }
-
-        if self.ch == Some('\n') {
-            // Unicode Character 'SYMBOL FOR NEWLINE' (U+2424)
-            return error!(ErrorKind::ExpectedToken { token: '\u{2424}' });
-        }
-
-        error!(ErrorKind::ExpectedToken { token: ch })
-    }
-
-    fn expect_indent(&mut self) -> Result<()> {
-        self.expect_char('\n')?;
-        self.skip_blank_lines();
-        self.expect_char(' ')?;
-        self.skip_inline_ws();
+        self.ptr += 1;
         Ok(())
     }
 
-    fn take_char_if(&mut self, ch: char) -> bool {
-        if self.ch == Some(ch) {
-            self.next();
-            return true;
+    pub fn take_if(&mut self, b: u8) -> bool {
```

Comment:
Nit: Perhaps rename this to `take_byte` or `consume_byte` for symmetry with `expect_byte`?

Reply:
`expect_*` is a fallible method that returns an error. I'll rename it to `take_byte_if`.